### PR TITLE
Buildcache relocate.py error fix

### PR DESCRIPTION
**Pull request by gartung**
Addresses  this error

> spack buildcache create -raf flecsi
---Lots of operations complete---
==> creating binary cache file for package cmake@3.14.2%gcc@7.4.0~doc+ncurses+openssl+ownlibs~qt arch=linux-centos7-x86_64 
==> Error: need more than 1 value to unpack

### CODE REVIEW

**Comment by odoublewen** [comment]
Hi -- For some reason, this recent commit breaks my build of `libxpm@3.5.12`.  (I have a CI job that rebuilds my spack packages every night, so I can detect breaks like this right away.)

This is on Centos7 with gcc 8.2.0.

```
==> Installing libxpm                                                                                                                                                  
==> Searching for binary cache of libxpm                                                                                                                               
==> Finding buildcaches in /bifx/apps/spack/mirror/build_cache                                                                                                         
==> No binary for libxpm found: installing from source                                                                                                                 
==> Using cached archive: /home/osolberg/temp/spack/var/spack/cache/libxpm/libxpm-3.5.12.tar.gz                                                                        
==> Staging archive: /home/osolberg/temp/spack/var/spack/stage/libxpm-3.5.12-dnmklmoiryh32rvh5h77xmw3475kq34w/libXpm-3.5.12.tar.gz                                     
==> Created stage in /home/osolberg/temp/spack/var/spack/stage/libxpm-3.5.12-dnmklmoiryh32rvh5h77xmw3475kq34w                                                          
==> No patches needed for libxpm                                                                                                                                       
==> Building libxpm [AutotoolsPackage]                                                                                                                                 
==> Executing phase: 'autoreconf'                                                                                                                                      
==> Executing phase: 'configure'                                                                                                                                       
==> Executing phase: 'build'                                                                                                                                           
==> Error: ProcessError: Command exited with status 2:                                                                                                                 
    'make' '-j16'                                                                                                                                                      
                                                                                                                                                                       
12 errors found in build log:                                                                                                                                          
     237             ^                                                                                                                                                 
     238    /home/osolberg/temp/spack/var/spack/stage/libxpm-3.5.12-dnmklmoiryh32rvh5h77xmw3475kq34w/libXpm-3.5.12/src/misc.c:100:9: warning: return discards 'const'  
            qualifier from pointer target type [-Wdiscarded-qualifiers]                                                                                                
     239      return ("Invalid XpmError");                                                                                                                             
     240             ^                                                                                                                                                 
     241      CCLD     cxpm                                                                                                                                            
     242    cxpm.o: In function `ErrorMessage':                                                                                                                        
  >> 243    /tmp/osolberg/spack-stage/spack-stage-aXno5h/libXpm-3.5.12/cxpm/cxpm.c:105: undefined reference to `libintl_gettext'                                       
  >> 244    /tmp/osolberg/spack-stage/spack-stage-aXno5h/libXpm-3.5.12/cxpm/cxpm.c:121: undefined reference to `libintl_gettext'                                       
  >> 245    /tmp/osolberg/spack-stage/spack-stage-aXno5h/libXpm-3.5.12/cxpm/cxpm.c:116: undefined reference to `libintl_gettext'                                       
  >> 246    /tmp/osolberg/spack-stage/spack-stage-aXno5h/libXpm-3.5.12/cxpm/cxpm.c:110: undefined reference to `libintl_gettext'                                       
  >> 247    /tmp/osolberg/spack-stage/spack-stage-aXno5h/libXpm-3.5.12/cxpm/cxpm.c:128: undefined reference to `libintl_gettext'                                       
     248    cxpm.o:/tmp/osolberg/spack-stage/spack-stage-aXno5h/libXpm-3.5.12/cxpm/cxpm.c:132: more undefined references to `libintl_gettext' follow                   
     249    cxpm.o: In function `main':                                                                                                                                
  >> 250    /tmp/osolberg/spack-stage/spack-stage-aXno5h/libXpm-3.5.12/cxpm/cxpm.c:149: undefined reference to `libintl_bindtextdomain'                                
  >> 251    /tmp/osolberg/spack-stage/spack-stage-aXno5h/libXpm-3.5.12/cxpm/cxpm.c:150: undefined reference to `libintl_textdomain'                                    
  >> 252    /tmp/osolberg/spack-stage/spack-stage-aXno5h/libXpm-3.5.12/cxpm/cxpm.c:157: undefined reference to `libintl_gettext'                                       
  >> 253    collect2: error: ld returned 1 exit status                                                                                                                 
  >> 254    make[2]: *** [cxpm] Error 1                                                                                                                                
     255    make[2]: Leaving directory `/tmp/osolberg/spack-stage/spack-stage-aXno5h/libXpm-3.5.12/cxpm'                                                               
  >> 256    make[1]: *** [all-recursive] Error 1                                                                                                                       
     257    make[1]: Leaving directory `/tmp/osolberg/spack-stage/spack-stage-aXno5h/libXpm-3.5.12'                                                                    
  >> 258    make: *** [all] Error 2                                                                                                                                    
                                                                                                                                                                       
See build log for details:                                                                                                                                             
  /home/osolberg/temp/spack/var/spack/stage/libxpm-3.5.12-dnmklmoiryh32rvh5h77xmw3475kq34w/spack-build.out    
```

If I do a `git checkout 964a1d5^` (to get the commit before this merge, namely, a777697), then the package `libxpm@3.5.12` builds without error.

I don't have time right now to investigate, or even open a formal issue, but thought I would make this known here in case the issue might be simply resolved.  I can provide more details later this week if needed.  Thanks!

File: var/spack/repos/builtin/packages/libxpm/package.py [hunk 1]
```diff
@@ -26,6 +26,6 @@ class Libxpm(AutotoolsPackage):
     depends_on('pkgconfig', type='build')
     depends_on('util-macros', type='build')
 
-    def setup_environment(self, spack_env, run_env):
-        spack_env.append_flags('LDFLAGS', '-L{0} -lintl'.format(
-            self.spec['gettext'].prefix.lib))
+#    def setup_environment(self, spack_env, run_env):
```
Comment (odoublewen) [inline var/spack/repos/builtin/packages/libxpm/package.py]: My error is due to these three lines being commented out.  Was this intentional?

Comment (gartung) [reply var/spack/repos/builtin/packages/libxpm/package.py]: Err... Sorry I don't know how that slipped in. I will submit a PR to fix this ASAP.

Comment (odoublewen) [reply var/spack/repos/builtin/packages/libxpm/package.py]: Thanks!

**Pull request merged**